### PR TITLE
fix(tearsheet): make tearsheet layout more dynamic 

### LINKED ISSUE #2403: [TearSheet] Dynamically added content breaks wrapper
### What package is this for?

- [X] React
- [ ] Angular

### Describe the bug

When content is added to header programmatically it is busting out the content section from its container. 

copied from #2390 

>**Summary**
>- This fixes an issue where if a content were dynamically added to the TearSheet header, the TearSheet content would expand beyond its container and get cropped by the bottom of the viewport.

### PR DESCRIPTION
**Pull request by davidicus**
Closes #2403 

**Summary**

Copied from #2390 

**Summary**
- This fixes an issue where if a content were dynamically added to the TearSheet header, the TearSheet content would expand beyond its container and get cropped by the bottom of the viewport.

**Acceptance Test (how to verify the PR)**
- Check the component story. I've added buttons on the tear sheets to include additional content to their header. The tear sheet size should be readjusted to prevent it from being cropped.

**Regression Test (how to make sure this PR doesn't break old functionality)**
The tests below can be checked via story
- Check the stack feature (one tear sheet on top of the other) to see if they still behave like before. 
- Check if the tear sheet's whole content is displayed

### CODE REVIEW

File: packages/react/src/components/TearSheet/TearSheetWrapper.jsx [hunk 1]
```diff
@@ -17,27 +16,22 @@ const propTypes = {
...
   className: undefined,
   onCloseAllTearSheets: undefined,
   children: undefined,
 };
 
+const baseClassName = `${iotPrefix}--tear-sheet-wrapper`;
+
 const TearSheetWrapper = ({ isOpen, className, onCloseAllTearSheets, children }) => {
-  const windowSize = useWindowSize();
+  // const windowSize = useWindowSize();
```
Comment (bjornalm) [inline packages/react/src/components/TearSheet/TearSheetWrapper.jsx:29]: Should we remove or uncomment these lines?

**Comment by bjornalm** [comment]
Perhaps not related to this PR but the new buttons described needs some margins in RTL
image

**Review by bjornalm — APPROVED** [review]
LGTM except for the commented code

File: packages/react/src/components/TearSheet/TearSheetWrapper.jsx [hunk 2]
```diff
@@ -17,27 +16,22 @@ const propTypes = {
...
   const [activeTearSheetIdx, setActiveTearSheetIdx] = useState(null);
   const childrenArray = Children.toArray(children).slice(0, 2); // Limit of 2 TearSheets
   const [containers] = useState(childrenArray.map(() => createRef()));
-  const [initialContainersWidth, setInitialContainersWidth] = useState(null);
-  const [containersStyles, setContainersStyles] = useState({});
+  // const [initialContainersWidth, setInitialContainersWidth] = useState(null);
+  // const [containersStyles, setContainersStyles] = useState({});
```
Comment (kevinsperrine) [inline packages/react/src/components/TearSheet/TearSheetWrapper.jsx:34]: more comments.

**Comment by davidicus** [comment]
> Perhaps not related to this PR but the new buttons described needs some margins in RTL
> image

I fixed. Thanks for the check.

File: packages/react/src/components/TearSheet/TearSheetWrapper.jsx (hunk 1 above)
Comment (davidicus) [reply packages/react/src/components/TearSheet/TearSheetWrapper.jsx]: removed comments

File: packages/react/src/components/TearSheet/TearSheetWrapper.jsx (hunk 2 above)
Comment (davidicus) [reply packages/react/src/components/TearSheet/TearSheetWrapper.jsx]: removed comments

**Pull request merged**